src/Mvc/Mvc.Core/src/Infrastructure/PhysicalFileResultExecutor.cs

```diff
@@ -86,28 +87,19 @@ protected virtual Task WriteFileAsync(ActionContext context, PhysicalFileResult
                 Logger.WritingRangeToBody();
             }
 
-            var sendFile = response.HttpContext.Features.Get<IHttpSendFileFeature>();
```

Comment:
This shouldn't need to change (I don't mind changing it) but we should make sure things are compatible when you try to get the old feature.

Reply:
Now the fallback is implemented at the server level so MVC doesn't need to check if the feature exists. I think we're better off Obsoleting (error) IHttpSendFileFeature. Otherwise every wrapper has to double assign itself in the feature collection.

Reply:
I guess the feature was optional anyways. I wonder if we should add a CanSendFile API that our extension method can use to check if it should polyfill. The problem with everything on the same interface is that they are all require by default now it’s not possible to tell if a server can implement optimized send file or not. I think this one needs to be left as a separate optional feature

Reply:
Except it's not really optional. If it was implemented by the server then it would bypass all response body shims. If it's not implemented then the fallback is required anyways.

I've centralized the fallback code so it's a one liner for anyone implementing the feature.

Reply:
OK that's fair. It means the SendFileAsync extension method could really be an instance method now. Not saying we need to change it but it's no longer optional.